lm_eval/api/task.py

```diff
@@ -88,24 +88,13 @@ class TaskConfig(dict):
     metadata: str = None  # by default, not used in the code. allows for users to pass arbitrary info to tasks
 
     def __post_init__(self):
-        # allow user-specified aliases so that users can
-        # force prompt-compatibility for some prompt regardless of
-        # field names in prompt
-        if self.template_aliases:
-            if type(self.doc_to_text) == str:
-                self.doc_to_text = self.template_aliases + self.doc_to_text
-
-            if type(self.doc_to_target) == str:
-                self.doc_to_target = self.template_aliases + self.doc_to_target
-
-            if type(self.gold_alias) == str:
-                self.gold_alias = self.template_aliases + self.gold_alias
 
         if self.generation_kwargs is not None:
             if self.output_type != "greedy_until":
                 eval_logger.warning(
-                    "passed `generation_kwargs`, but not using a generation request type!"
+                    "passed `generation_kwargs`, but not using `output_type: greedy_until`!"
                 )
+                assert self.output_type != "greedy_until"
```

Comment:
@haileyschoelkopf This assert does nothing. Check `self.output_type != "greedy_until"` is always True inside this conditional part. So no AssertionError. If it was supposed to raise error and stop processing, then something like `raise ValueError("passed `generation_kwargs`, but not using `output_type: greedy_until`!")` could be used.

Reply:
Would you be willing to make this change in #1533 ?

Reply:
I will raise ValueError with message from logger (logger call removed). Is that OK?

Reply:
yup, SGTM!

Reply:
Done (also in other files from #1533)!